hadoop-ozone/common/src/main/proto/OzoneManagerProtocol.proto

```diff
@@ -1097,11 +1097,34 @@ message UpdateGetS3SecretRequest {
     required string awsSecret = 2;
 }
 
+message OMServiceId {
+    required string serviceID = 1;
+}
+
+/**
+  This proto is used to define the OM node Id and its ratis server state.
+*/
+message RoleInfo {
+    required string omNodeID = 1;
+    required string ratisServerRole = 2;
+}
+
+/**
+  This is used to get the Server States of OMs.
+*/
+message ServiceState {
+    repeated RoleInfo roleInfos = 1;
+}
+
 /**
  The OM service that takes care of Ozone namespace.
 */
 service OzoneManagerService {
     // A client-to-OM RPC to send client requests to OM Ratis server
     rpc submitRequest(OMRequest)
           returns(OMResponse);
+
+    // A client-to-OM RPC to get ratis server states of OMs
+    rpc getServiceState(OMServiceId)
+          returns(ServiceState);
 }
```

Comment:
What is the use case for this RPC? it is used only from the OM HA tool? or is some thing that clients need to know when communicating with the OM? if clients need to know this info, then clients already make a call called getServiceList and this info is more appropriate to be part of that call? Then both clients and tools like OM HA can get to that information pretty easily.

Reply:
Clients do not need this information. 
I have added a new OMRequest to get the OM server roles. Did not add it to getServiceList api because getServiceList is called when creating RpcClient (so all clients make this call when starting up). But adding the server roles also to this api would mean one extra Ratis call to get the server roles which is not required by all clients. Hence, kept the two calls separate.

Reply:
Sorry did not see this comment. How can a client communicate to OM? Does it not need to send the request to the leader ?